// @storybook/addon-vitest/dist/_node-chunks/chunk-BM6TFQ6X.js
import CJS_COMPAT_NODE_URL_vppyn5e0su from 'node:url';
import CJS_COMPAT_NODE_PATH_vppyn5e0su from 'node:path';
import CJS_COMPAT_NODE_MODULE_vppyn5e0su from "node:module";

var __filename = CJS_COMPAT_NODE_URL_vppyn5e0su.fileURLToPath(import.meta.url);
var __dirname = CJS_COMPAT_NODE_PATH_vppyn5e0su.dirname(__filename);
var require = CJS_COMPAT_NODE_MODULE_vppyn5e0su.createRequire(import.meta.url);

// ------------------------------------------------------------
// end of CJS compatibility banner, injected by Storybook's esbuild configuration
// ------------------------------------------------------------

// ../../core/src/component-testing/constants.ts
var ADDON_ID = "storybook/interactions";
var PANEL_ID = `${ADDON_ID}/panel`;
var DOCUMENTATION_LINK = "writing-tests/integrations/vitest-addon";
var DOCUMENTATION_DISCREPANCY_LINK = `${DOCUMENTATION_LINK}#what-happens-when-there-are-different-test-results-in-multiple-environments`;

// ../a11y/src/constants.ts
var ADDON_ID2 = "storybook/a11y";
var PANEL_ID2 = `${ADDON_ID2}/panel`;
var UI_STATE_ID = `${ADDON_ID2}/ui`;
var RESULT = `${ADDON_ID2}/result`;
var REQUEST = `${ADDON_ID2}/request`;
var RUNNING = `${ADDON_ID2}/running`;
var ERROR = `${ADDON_ID2}/error`;
var MANUAL = `${ADDON_ID2}/manual`;
var SELECT = `${ADDON_ID2}/select`;
var DOCUMENTATION_LINK2 = "writing-tests/accessibility-testing";
var DOCUMENTATION_DISCREPANCY_LINK2 = `${DOCUMENTATION_LINK2}#why-are-my-tests-failing-in-different-environments`;

// src/constants.ts
var ADDON_ID3 = "storybook/test";
var TEST_PROVIDER_ID = `${ADDON_ID3}/test-provider`;
var STORYBOOK_ADDON_TEST_CHANNEL = "STORYBOOK_ADDON_TEST_CHANNEL";
var DOCUMENTATION_LINK3 = "writing-tests/integrations/vitest-addon";
var DOCUMENTATION_FATAL_ERROR_LINK = `${DOCUMENTATION_LINK3}#what-happens-if-vitest-itself-has-an-error`;
var COVERAGE_DIRECTORY = "coverage";
var SUPPORTED_FRAMEWORKS = [
  "@storybook/nextjs",
  "@storybook/nextjs-vite",
  "@storybook/react-vite",
  "@storybook/preact-vite",
  "@storybook/svelte-vite",
  "@storybook/vue3-vite",
  "@storybook/html-vite",
  "@storybook/web-components-vite",
  "@storybook/sveltekit",
  "@storybook/react-native-web-vite"
];
var storeOptions = {
  id: ADDON_ID3,
  initialState: {
    config: {
      coverage: false,
      a11y: false
    },
    watching: false,
    cancelling: false,
    fatalError: void 0,
    indexUrl: void 0,
    previewAnnotations: [],
    currentRun: {
      triggeredBy: void 0,
      config: {
        coverage: false,
        a11y: false
      },
      componentTestCount: {
        success: 0,
        error: 0
      },
      a11yCount: {
        success: 0,
        warning: 0,
        error: 0
      },
      storyIds: void 0,
      totalTestCount: void 0,
      startedAt: void 0,
      finishedAt: void 0,
      unhandledErrors: [],
      coverageSummary: void 0
    }
  }
};
var STORE_CHANNEL_EVENT_NAME = `UNIVERSAL_STORE:${storeOptions.id}`;
var STATUS_STORE_CHANNEL_EVENT_NAME = "UNIVERSAL_STORE:storybook/status";
var TEST_PROVIDER_STORE_CHANNEL_EVENT_NAME = "UNIVERSAL_STORE:storybook/test-provider";
var STATUS_TYPE_ID_COMPONENT_TEST = "storybook/component-test";
var STATUS_TYPE_ID_A11Y = "storybook/a11y";

export {
  ADDON_ID3 as ADDON_ID,
  STORYBOOK_ADDON_TEST_CHANNEL,
  DOCUMENTATION_LINK3 as DOCUMENTATION_LINK,
  COVERAGE_DIRECTORY,
  SUPPORTED_FRAMEWORKS,
  storeOptions,
  STORE_CHANNEL_EVENT_NAME,
  STATUS_STORE_CHANNEL_EVENT_NAME,
  TEST_PROVIDER_STORE_CHANNEL_EVENT_NAME,
  STATUS_TYPE_ID_COMPONENT_TEST,
  STATUS_TYPE_ID_A11Y
};
